docs/intro/overrides.rst

```diff
@@ -0,0 +1,349 @@
+      something like:
+
+      - :sub:`(match) https://www.dualexample.com/shop/electronics/?product=123`
+      - :sub:`(match) https://www.dualexample.com/shop/books/paperback/?product=849`
+      - :sub:`(NO match) https://www.dualexample.com/on-sale/books/?product=923`
+      - :sub:`(match) https://www.dualexample.net/store/kitchen/?pid=776`
+      - :sub:`(match) https://www.dualexample.net/store/?pid=892`
+      - :sub:`(NO match) https://www.dualexample.net/new-offers/fitness/?pid=892`
+
+    - On the other hand, ``AnotherExampleProductPage`` is only used instead of
+      ``GenericProductPage`` when we're handling pages from ``anotherexample.com``
+      that doesn't contain ``/digital-goods/`` in its URL path.
+
+.. tip::
+
+    The URL patterns declared in the ``@handle_urls`` annotation can still be
+    further customized. You can read some of the specific parameters in the
+    :ref:`API section <api-overrides>` of :func:`web_poet.handle_urls`.
+
+
+Retrieving all available Overrides
+----------------------------------
+
+The :meth:`~.PageObjectRegistry.get_overrides` method from the ``web_poet.default_registry``
+allows discovery and retrieval of  all :class:`~.OverrideRule` from your project.
+Following from our example above, using it would be:
+
+.. code-block:: python
+
+    from web_poet import default_registry
+
+    # Retrieves all OverrideRules that were registered in the registry
+    rules = default_registry.get_overrides()
+
+    print(len(rules))  # 3
+    print(rules[0])    # OverrideRule(for_patterns=Patterns(include=['example.com'], exclude=[], priority=500), use=<class 'my_project.page_objects.ExampleProductPage'>, instead_of=<class 'my_project.page_objects.GenericProductPage'>, meta={})
+
+Remember that using ``@handle_urls`` to annotate the Page Objects would result
+in the :class:`~.OverrideRule` to be written into ``web_poet.default_registry``.
+
+
+.. warning::
+
+    :meth:`~.PageObjectRegistry.get_overrides` relies on the fact that all essential
+    packages/modules which contains the :func:`web_poet.handle_urls`
+    annotations are properly loaded.
+
+    Thus, for cases like importing Page Objects from another external package,
+    you'd need to properly load all ``@handle_urls`` annotations from the external
+    source. This ensures that the external Page Objects have their annotations
+    properly loaded.
+
+    This can be done via the function named :func:`~.web_poet.overrides.consume_modules`.
+    Here's an example:
+
+    .. code-block:: python
+
+        from web_poet import default_registry, consume_modules
+
+        consume_modules("external_package_A.po", "another_ext_package.lib")
+        rules = default_registry.get_overrides()
+
+        # Fortunately, `get_overrides()` provides a shortcut for the lines above:
+        rules = default_registry.get_overrides(consume=["external_package_A.po", "another_ext_package.lib"])
```

Comment:
Because of the caveats I'm not sure it makes sense to provide `consume` keyword for get_overrides - the code above reads as it'd return only POs from the listed modules. With 

```py
consume_modules("external_package_A.po", "another_ext_package.lib")
rules = default_registry.get_overrides()
```

the caveat is more clear IMHO, and it's the same amount of code. Do you think we can remove this method?

Reply:
Yeah, we can make it more explicit by having to access `consume_modules()` externally.

Removed this on https://github.com/scrapinghub/web-poet/pull/27/commits/37e5ba1a3696f629801478fe78ea1729164f39df